progs/fasta-recs/main.go

```diff
@@ -18,5 +20,32 @@ func main() {
 	}
 	defer f.Close()
 
-	fmt.Println("I should be processing the input file now!")
+	//###########################################
+
+	output := ""
+	fileScanner := bufio.NewScanner(f)
+
+	//scan file line by line
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+
+		if len(line) == 0 {
+			continue
+		}
+
+		//handle 'name of sequence' cases
+		if line[0:1] == ">" {
+			if output != "" {
+				output = output + "\n"
+			}
+			output = output + strings.TrimSpace(line[1:]) + "	"
+
+			//handle 'sequence' cases
+		} else {
+			output = output + line
+		}
+	}
```

Comment:
This is a functional implementation, but string concatenation involves copying both strings, so you have a quadratic time solution here.

From Go 1.10+ you have `strings.Builder` that can handle it more efficiently, giving you a linear time solution. Alternatively, you can collect the sequence lines in a slice and join them at the end. That will also be linear.